muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/form-element.tsx

```diff
@@ -160,23 +164,28 @@ export class SearchFormElement extends React.Component<SearchFormElementProps, S
 
   constructor(props: SearchFormElementProps) {
     super(props);
-    this.state = {
-      value: this.props.value
-    }
     this.updateSearchField = this.updateSearchField.bind(this);
+    this.clearSearchField = this.clearSearchField.bind(this);
   }
 
   updateSearchField(e: React.ChangeEvent<HTMLInputElement>) {
     let value = e.target.value;
     this.props.updateField(value);
   }
 
+  clearSearchField() {
+    this.props.updateField("");
+    document.getElementById(this.props.id).focus();
+  }
+
   render() {
     const modifiers = this.props.modifiers && this.props.modifiers instanceof Array ? this.props.modifiers : [this.props.modifiers];
     return (
-      <div className="form-element form-element--coursepicker-toolbar">
-        <input name={this.props.name} value={this.state.value} className="form-element__input form-element__input--main-function-search" placeholder={this.props.placeholder} onChange={this.updateSearchField} />
-        <div className="form-element__input-decoration--main-function-search icon-search"></div>
+      <div className={`form-element form-element--search ${this.props.modifiers ? modifiers.map(m => `form-element--${m}`).join(" ") : ""} ${this.props.className ? this.props.className : ""}`}>
+        <label htmlFor={this.props.id} className="visually-hidden">{this.props.placeholder}</label>
+        <input id={this.props.id} onFocus={this.props.onFocus} onBlur={this.props.onBlur} name={this.props.name} value={this.props.value} className={`form-element__input form-element__input--search ${this.props.modifiers ? modifiers.map(m => `form-element__input--${m}`).join(" ") : ""}`} placeholder={this.props.placeholder} onChange={this.updateSearchField} />
+        <div className={`form-element__input-decoration--clear-search icon-cross ${this.props.value.length > 0 ? "active" : ""}`} onClick={this.clearSearchField}></div>
+        <div className="form-element__input-decoration--search icon-search"></div>
```

Comment:
We should transform **search** and **clear search** elements to buttons  in the future and add visuaally-hidden labels to them so we can then provide support for keyboard users as well within search fields. If we use buttons instead of div elements then keybords tabindex will work automatically, hidden label with proper `id` and `htmlFor` attributes then provides information to the screen readers as well.

Also `icon-search` element could have some sort of event tied to it so it can actually work as an search button as well even if we don't require the button press to activate search currently. We need to investigate what accessibility directive specifies in those cases where search is not trigger via button press but instead is automatic based on users typing intervals.